Library/Homebrew/os/linux/elf.rb

```diff
@@ -43,7 +43,7 @@ def initialize(path)
       else
         odie "patchelf must be installed: brew install patchelf"
       end
-      if needed.empty?
+      if needed.empty? || path.arch != Hardware::CPU::arch
```

Comment:
I've merged PR https://github.com/Linuxbrew/brew/pull/370 for `Hardware::CPU.arch`